### PR TITLE
[FIXED JENKINS-37189] Only append `?uctest` to http/https update center URLs

### PR DESCRIPTION
**Pull request by stephenc**
- Also fixes to append `&uctest` if the update center URL already contains a query fragment

@jenkinsci/code-reviewers @reviewbybees

@olivergondza can we get an exemption so that this can be included in the backports for 2.7.3?

### CODE REVIEW

File: core/src/main/java/hudson/model/UpdateCenter.java [hunk 1]
```diff
@@ -1042,7 +1042,11 @@ public void checkConnection(ConnectionCheckJob job, String connectionCheckUrl) t
          * @throws IOException if a connection to the update center server can't be established.
          */
         public void checkUpdateCenter(ConnectionCheckJob job, String updateCenterUrl) throws IOException {
-            testConnection(new URL(updateCenterUrl + "?uctest"));
+            if (updateCenterUrl.startsWith("http://") || updateCenterUrl.startsWith("https://")) {
+                testConnection(new URL(updateCenterUrl + (updateCenterUrl.indexOf('?') == -1 ? "?uctest" : "&uctest")));
```
Comment (jglick) [inline core/src/main/java/hudson/model/UpdateCenter.java]: :ant: `String.contains`

**Comment by amuniz** [comment]
:bee:

**Comment by recena** [comment]
🐝

**Comment by jglick** [comment]
:bee:

**Comment by jtnord** [comment]
:bee:  /cc @andresrc

File: core/src/main/java/hudson/model/UpdateCenter.java (hunk 1 above)
Comment (stephenc) [inline core/src/main/java/hudson/model/UpdateCenter.java]: less performant than `indexOf(character)`

**Comment by stephenc** [comment]
@reviewbybees done

File: core/src/main/java/hudson/model/UpdateCenter.java (hunk 1 above)
Comment (daniel-beck) [inline core/src/main/java/hudson/model/UpdateCenter.java]: How often do you expect this code to be called per second?

**Comment by oleg-nenashev** [comment]
Oh well, PR builder...

```
12:46:18 [ERROR] Failed to execute goal org.codehaus.mojo:animal-sniffer-maven-plugin:1.15:check (default) on project pom: Execution default of goal org.codehaus.mojo:animal-sniffer-maven-plugin:1.15:check failed: A required class was missing while executing org.codehaus.mojo:animal-sniffer-maven-plugin:1.15:check: org/objectweb/asm/ClassVisitor
12:46:18 [ERROR] -----------------------------------------------------
```

**Comment by oleg-nenashev** [comment]
🐝

File: core/src/main/java/hudson/model/UpdateCenter.java (hunk 1 above)
Comment (stephenc) [inline core/src/main/java/hudson/model/UpdateCenter.java]: :-P

**Comment by andresrc** [comment]
:bee:

**Comment by olivergondza** [comment]
No unittest?

**Comment by stephenc** [comment]
@olivergondza happy now? ;-)

**Comment by olivergondza** [comment]
:+1:

**Pull request merged**